contracts/src/TeleporterTokenSource.sol

```diff
@@ -240,11 +255,17 @@ abstract contract TeleporterTokenSource is
         } else if (bridgeMessage.messageType == BridgeMessageType.MULTI_HOP_SEND) {
             MultiHopSendMessage memory payload =
                 abi.decode(bridgeMessage.payload, (MultiHopSendMessage));
+
```

Comment:
I think we need to scale the `payload.secondaryFee` amount here and in the multi-hop call case prior to using it as the `primaryFee` amount of the second message.

Also the wording in these comments is a bit confused/may have a typo ("passed.abi")

Reply:
Should be done now, I'll add tests when I get a chance